Streetcode/Streetcode.XUnitTest/MediatRTests/StreetCode/Streetcode/CreateStreetcodeTests.cs

```diff
@@ -178,5 +179,35 @@ public async Task AddImagesAsync_ValidImagesIds_DoesNotThrowException()
             Func<Task> action = async () => await handler.AddImagesAsync(streetcode, imagesIds);
             await action.Should().NotThrowAsync<HttpRequestException>();
         }
+        
+        [Theory]
+        [InlineData(-5)]
+        [InlineData(10000000)]
+        public async Task Handle_IndexIsOutOfBounds_ReturnsFailure(int index)
+        {
+            // Arrange
+            StreetcodeCreateDTO streetcode = new StreetcodeCreateDTO() { Index = index };
+            var mapperMock = new Mock<IMapper>();
+            var repositoryWrapperMock = new Mock<IRepositoryWrapper>();
+            repositoryWrapperMock.Setup(repo => repo.StreetcodeImageRepository.CreateRangeAsync(It.IsAny<IEnumerable<StreetcodeImage>>()))
+                .Returns(Task.CompletedTask);
+            var loggerServiceMock = new Mock<ILoggerService>();
+            var stringLocalizerAnErrorOccurredMock = new Mock<IStringLocalizer<AnErrorOccurredSharedResource>>();
+            var stringLocalizerFailedToCreateMock = new Mock<IStringLocalizer<FailedToCreateSharedResource>>();
+
```

Comment:
This chunk of code is very hard to read.
Pay attention to the whole file - the same mocks are used by many tests. You can try to consolidate common setup code in a shared setup method.
See some better examples in unit tests for other functional units (e.g. News)